echopop/extensions/diagnostics.py

```diff
@@ -0,0 +1,510 @@
+
+
+class DiagnosticPlot:
+    """
+    Echopop diagnostic plots used for visually scrutinizing inputs and results
+
+    This class includes visualizations used by the FEAT team
+
+    Parameters
+    ----------
+    survey: echopop.survey.Survey
+        An `echopop.survey.Survey` class object containing all of the relevant results required for
+        producing various `*.xlsx`-formatted reports
+
+    Attributes
+    ----------
+    data: echopop.survey.Survey
+        The `echopop.survey.Survey`-class object containing all of the data used to generate the
+        requested diagnostic visualizations
+    tile: geoviews.elements.geo.WMTS
+        A GeoViews tile element comprising an ESRI ocean basemap for plotting
+    """
+
+    # Class-level flag to track renderer initialization
+    _is_bokeh_initialized = False
+
+    # Class-level flag to track renderer initialization
+    _is_theme_initialized = False
+
+    def __init__(self, survey: Survey):
+
+        # Evaluate whether `bokeh` has been globally initialized
+        if not DiagnosticPlot._is_bokeh_initialized:
+            # ---- Initialize the extension
+            hv.extension("bokeh")
+            # ---- Mark the class flag to prevent redundant initializations
+            DiagnosticPlot._is_bokeh_initialized = True
+
+        # Evaluate whether `bokeh.theme` has been globally initialized
+        if not DiagnosticPlot._is_theme_initialized:
+            # ---- Initialize the theme
+            hv.renderer("bokeh").theme = bok.themes.Theme(json=self._json_theme)
+
+        # Assign the data
+        self.data = survey
+
+        # Store the ESRI baselayer map to avoid redundant calls
+        self.tile = gv.tile_sources.EsriOceanBase
+
+    def mesh_cropping_results(self) -> pn.Column:
+
+        # Get the datasets
+        DATAINPUT = {
+            "Latitude/longitude": {
+                "cropped": (
+                    self.data.analysis["kriging"]["mesh_df"].copy()
+                    if "kriging" in self.data.analysis
+                    else {}
+                ),
+                "full": self.data.input["statistics"]["kriging"]["mesh_df"].copy(),
+            },
+            "Isobath-referenced": {
+                "cropped": self.data.analysis["kriging"]["mesh_df"].copy(),
+            },
+        }
+
+        # Get the isobath data
+        isobath_df = self.data.input["statistics"]["kriging"]["isobath_200m_df"]
+
+        # Amend the column names, if needed
+        [
+            v.rename(
+                columns={
+                    col: (
+                        "longitude"
+                        if "longitude" in col
+                        else "latitude" if "latitude" in col else col
+                    )
+                    for col in v.columns
+                },
+                inplace=True,
+            )
+            for k, v in DATAINPUT["Latitude/longitude"].items()
+        ]
```

Comment:
Since this file `self.data.input["statistics"]["kriging"]["isobath_200m_df"]` is part of Echopop standard input, I think it's better to force the excel file/datafram to be of a specific format instead of doing this type of automatic 
 and "silent" reformatting. I am not a big proponent of silent reformatting in anticipation of user errors -- I think people need to know what they put into the analysis.